Content.Server/Tools/Innate/InnateToolSystem.cs

```diff
@@ -1,90 +1,100 @@
+///     borgs, or maybe even stuff like changeling armblades!
+/// </summary>
+public sealed class InnateToolSystem : EntitySystem
 {
-    /// <summary>
-    ///     Spawns a list unremovable tools in hands if possible. Used for drones,
-    ///     borgs, or maybe even stuff like changeling armblades!
-    /// </summary>
-    public sealed class InnateToolSystem : EntitySystem
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly SharedHandsSystem _sharedHandsSystem = default!;
+    [Dependency] private readonly TagSystem _tagSystem = default!;
+
+    private List<string> _toSpawn = new();
+
+    public override void Initialize()
     {
-        [Dependency] private readonly IRobustRandom _robustRandom = default!;
-        [Dependency] private readonly SharedHandsSystem _sharedHandsSystem = default!;
-        [Dependency] private readonly TagSystem _tagSystem = default!;
-        public override void Initialize()
-        {
-            base.Initialize();
-            SubscribeLocalEvent<InnateToolComponent, ComponentStartup>(OnStartup);
-            SubscribeLocalEvent<InnateToolComponent, ComponentShutdown>(OnShutdown);
-            SubscribeLocalEvent<InnateToolComponent, DestructionEventArgs>(OnDestroyed);
-        }
+        base.Initialize();
+        SubscribeLocalEvent<InnateToolComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<InnateToolComponent, BodyPartAddedEvent>(OnPartAdded);
+        SubscribeLocalEvent<InnateToolComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<InnateToolComponent, DestructionEventArgs>(OnDestroyed);
+    }
 
-        private void OnStartup(EntityUid uid, InnateToolComponent component, ComponentStartup args)
-        {
-            if (component.Tools.Count == 0)
-                return;
+    private void OnMapInit(EntityUid uid, InnateToolComponent component, MapInitEvent args)
+    {
+        if (component.Tools.Count == 0)
+            return;
 
-            var spawnCoord = Transform(uid).Coordinates;
+        _toSpawn = EntitySpawnCollection.GetSpawns(component.Tools, _robustRandom);
+    }
 
-            if (TryComp<HandsComponent>(uid, out var hands) && hands.Count >= component.Tools.Count)
-            {
-                var items = EntitySpawnCollection.GetSpawns(component.Tools, _robustRandom);
-                foreach (var entry in items)
-                {
-                    var item = Spawn(entry, spawnCoord);
-                    AddComp<UnremoveableComponent>(item);
-                    if (!_sharedHandsSystem.TryPickupAnyHand(uid, item, checkActionBlocker: false))
-                    {
-                        QueueDel(item);
-                        continue;
-                    }
-                    component.ToolUids.Add(item);
-                }
-            }
-        }
+    private void OnPartAdded(EntityUid uid, InnateToolComponent component, BodyPartAddedEvent args)
```

Comment:
do `HandCountChangedEvent` rather than `BodyPartAddedEvent`

Reply:
Somehow I must've missed that event while looking for it, thanks